src/sentry/deletions/tasks/nodestore.py

```diff
@@ -0,0 +1,221 @@
+    """
+    if not group_ids:
+        raise DeleteAborted("delete_events_from_nodestore.empty_group_ids")
+
+    args_to_schedule_next_task = [
+        organization_id,
+        project_id,
+        group_ids,
+        transaction_id,
+        dataset_str,
+        referrer,
+    ]
+    dataset = Dataset(dataset_str)
+
+    # These can be used for debugging
+    extra = {"project_id": project_id, "transaction_id": transaction_id}
+    sentry_sdk.set_tags(extra)
+    logger.info("deletions.nodestore.started", extra=extra)
+    project = Project.objects.get(id=project_id)
+    assert project.organization_id == organization_id
+    task = NodestoreDeletionTask(
+        organization_id=organization_id,
+        project_id=project_id,
+        group_ids=group_ids,
+        dataset=dataset,
+        referrer=referrer,
+    )
+
+    try:
+        # Fetch events for deletion
+        events = task.fetch_events(last_event_id, last_event_timestamp)
+        if len(events) > 0:
+            last_event = events[-1]
+            _delete_events_from_nodestore(events=events, extra=extra)
+            _delete_dangling_attachments_and_user_reports(events, [project_id])
+            delete_events_for_groups_from_nodestore.apply_async(
+                args=args_to_schedule_next_task,
+                kwargs={
+                    "last_event_id": last_event.event_id,
+                    "last_event_timestamp": last_event.timestamp,
+                },
+            )
+        else:
+            logger.info("deletions.nodestore.completed", extra=extra)
+
+    except Exception:
+        metrics.incr("deletions.nodestore.delete_events_from_nodestore.error", 1, sample_rate=1)
+        logger.warning("deletions.nodestore.failed", extra=extra)
+        raise
+
+
+class NodestoreDeletionTask:
+    def __init__(
+        self,
+        organization_id: int,
+        project_id: int,
+        group_ids: Sequence[int],
+        dataset: Dataset,
+        referrer: str,
+    ):
+        """
+        Initialize the NodestoreDeletionTask.
+
+        Args:
+            organization_id: Organization ID for tenant context
+            project_id: Project ID that all groups belong to
+            group_ids: List of group IDs to delete events for
+            dataset: Dataset to delete events from
+            referrer: Referrer for the task
+        """
+        self.organization_id = organization_id
+        self.project_id = project_id
+        self.group_ids = group_ids
+        self.dataset = dataset
+        self.referrer = referrer
+
+    @property
+    def tenant_ids(self) -> Mapping[str, Any]:
+        return {"referrer": self.referrer, "organization_id": self.organization_id}
+
+    def fetch_events(
+        self,
+        last_event_id: str | None = None,
+        last_event_timestamp: str | None = None,
+    ) -> list[Event]:
+        """Fetch error events for the given groups."""
+        return _fetch_events(
```

Comment:
Do you need this class? It seems like a fancy wrapper around this function. You could call the wrapped function directly and have less complexity.